### PR TITLE
Add descriptive stat scripts, a test and derive main descriptive stats

### PR DESCRIPTION
**Pull request by kerembaskaya**

### CODE REVIEW

File: inflation/dataset/wrangler.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+FILE_DIR_PATH = Path(__file__).parents[2] / "inflation-resources/data/"
+
+
+def save_dataset(dataset: InflationDataset, output_file_name):
+    """
+    It saves InflationJSONA101DatasetReader records by serializing.
+    """
+    full_path = FILE_DIR_PATH / output_file_name
+
+    with open(full_path, "wb") as f:
```
Comment (osmanbaskaya) [inline inflation/dataset/wrangler.py:18]: This module can be dataset's methods, right? We can move all these to there

File: inflation/dataset/reader.py [hunk 2]
```diff
@@ -185,7 +187,7 @@ def read(self, filename, fields=None) -> InflationDataset:
                         date,
                     )
                     records.append(record)
-        print(total_pages, item_pages)
+        # print(total_pages, item_pages)
```
Comment (osmanbaskaya) [inline inflation/dataset/reader.py:190]: It is still useful to see total. Could you remove other comments but could you log.info the result at the end?

**Comment by osmanbaskaya** [comment]
You pushed two full pickled data. Could you remove one of them?

File: inflation/dataset/wrangler.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        cPickle.dump(dataset, f)
+
+
+def read_dataset(file_path):
```
Comment (osmanbaskaya) [inline inflation/dataset/wrangler.py:22]: class method (InflationDataset.load(...))

File: inflation/dataset/reader.py (hunk 2 above)
Comment (kerembaskaya) [reply inflation/dataset/reader.py:190]: I added it.

File: inflation-resources/tests/dataset/test_reader.py [hunk 4]
```diff
@@ -13,12 +11,23 @@
...
+    / "inflation-resources/tests/data/a101.med.json.gz"
+)
+
 
 @pytest.fixture(scope="function")
 def inflation_data_reader():
     return InflationJSONA101DatasetReader()
 
 
+# Q: Niye yine de her testte dunya kadar bekliyor?
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_reader.py:25]: not sure, let's debug together.

File: inflation-resources/tests/dataset/test_reader.py [hunk 5]
```diff
@@ -13,12 +11,23 @@
...
+@pytest.fixture(scope="module")
+def record_data():
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_reader.py:27]: change the name? it is dataset

File: inflation-resources/tests/dataset/test_reader.py [hunk 6]
```diff
@@ -13,12 +11,23 @@
...
+    return InflationJSONA101DatasetReader().read(A101_MED_DATA_PATH)
+
+
 def test_read_product(
     inflation_data_reader,
 ):  # Q: 3 tane yok muydu niye 2 tane geldi?
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_reader.py:33]: mind removing the comment?

File: inflation-resources/tests/dataset/test_reader.py [hunk 7]
```diff
@@ -48,3 +57,10 @@ def test_read_product(
 
     for record, truth in zip(record_data, truths):
         assert record == truth
+
+
+def test_length_readable(record_data):
+    """
+    It controls the actual number of product pages that are read.
+    """
+    assert len(record_data) == 206
```
Comment (osmanbaskaya) [inline inflation-resources/tests/dataset/test_reader.py:66]: maybe we should reduce the size of the data. we don't really use it.

File: inflation/dataset/reader.py [hunk 8]
```diff
@@ -1,12 +1,22 @@
...
+import _pickle as cPickle
+import pandas as pd
 
+from dataclasses import dataclass
+from pathlib import Path
 from bs4 import BeautifulSoup
 
+from common.customized_logging import configure_logging
+
+FILE_DIR_PATH = Path(__file__).parents[2] / "inflation-resources/data/"
```
Comment (osmanbaskaya) [inline inflation/dataset/reader.py:14]: name is not great.

File: inflation/dataset/reader.py [hunk 9]
```diff
@@ -185,15 +211,13 @@ def read(self, filename, fields=None) -> InflationDataset:
                         date,
                     )
                     records.append(record)
-        print(total_pages, item_pages)
+        logger.info(f"total_pages={total_pages}, item_pages={item_pages}")
```
Comment (osmanbaskaya) [inline inflation/dataset/reader.py:214]: 👍🏻

File: inflation/dataset/reader.py [hunk 10]
```diff
@@ -47,6 +57,23 @@ def __getitem__(self, s):
...
+
+        with open(full_path, "wb") as f:
+            cPickle.dump(self.dataset, f)
+
+    @classmethod
+    def read_dataset(cls, file_path):
+        with open(file_path, "rb") as input_file:
+            return cPickle.load(input_file)
+
+    def convert_dataset_to_dataframe(self):
```
Comment (osmanbaskaya) [inline inflation/dataset/reader.py:74]:
```suggestion
    def to_df(self):
```

File: inflation/analysis/descriptive-analysis.ipynb [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Save dataset\n",
+    "\n",
+    "dataset.save_dataset('A101_full_data.pickle1')"
```
Comment (osmanbaskaya) [inline inflation/analysis/descriptive-analysis.ipynb:51]: not a great naming. do not play with extensions.

**Comment by osmanbaskaya** [comment]
Let me know when you adress those.

File: inflation/dataset/reader.py (hunk 8 above)
Comment (kerembaskaya) [reply inflation/dataset/reader.py:14]: I changed as SAVED_DATASETS_DIR_PATH

Comment (osmanbaskaya) [reply inflation/dataset/reader.py:14]: DATASET_PATH is better.

File: inflation/analysis/descriptive-analysis.ipynb (hunk 11 above)
Comment (osmanbaskaya) [reply inflation/analysis/descriptive-analysis.ipynb:51]: duruyor

**Review by osmanbaskaya — APPROVED** [review]
LGTM

File: requirements.txt [hunk 12]
```diff
@@ -2,15 +2,15 @@
...
+beautifulsoup4~=4.10.0
+setuptools~=58.0.4
 scikit_learn==0.24.2
 urlpath==1.1.7
 dill==0.3.4
 gunicorn==20.1.0
 sklearn~=0.0
 scikit-learn~=0.24.2
-numpy~=1.21.1
+numpy~=1.20.3
```
Comment (osmanbaskaya) [inline requirements.txt:13]: Why downgrading?

**Comment by osmanbaskaya** [comment]
Can we create all the descriptive stats we thought it would be useful to have, and close this PR? Let's make everything functional, so that when we choose another website we can use the same code to get the descriptive stats for that dataset.

**Comment by osmanbaskaya** [comment]
are we not going to merge this?

**Pull request closed without merging**